protocols/kad/src/behaviour.rs

```diff
@@ -2433,14 +2433,17 @@ where
         // Run the periodic provider announcement job.
         if let Some(mut job) = self.add_provider_job.take() {
             let num = usize::min(JOBS_MAX_NEW_QUERIES, jobs_query_capacity);
+            let mut ready_provider_keys: Vec<Key> = Vec::new();
             for _ in 0..num {
                 if let Poll::Ready(r) = job.poll(cx, &mut self.store, now) {
-                    self.start_add_provider(r.key, AddProviderContext::Republish)
```

Comment:
Before we were calling `start_add_provider` for each one that was `Ready`, now we only do it for the first one. Isn't that wrong?

Reply:
You're right! Let me fix this.